### PR TITLE
✨ Added PollResponse API

### PR DESCRIPTION
**Pull request by PushkarDureja**
### Changes implemented in order to solve [#95](https://github.com/RocketMeet/RocketMeet-client/issues/95) 

- Created an API `/pollResponse` which is responsible for sending mails
- Created an email template

### CODE REVIEW

**Comment by AnandBaburajan** [comment]
Congrats and thanks! The failing tests are not your fault, by the way. We're looking into it.

**Comment by AthulJoshi** [comment]
@PushkarDureja Thanks for the PR.
But I have one concern, i.e we are sending the mail to the poll creator right? So is there a use for the receiverIDs.

**Comment by PushkarDureja** [comment]
You are right, receiverIds array will always be of length 1 in this case. I reused the MailerArg interface here because i thought creating a new interface or altering the interface for a single field is not effective in this case

**Comment by AthulJoshi** [comment]
Is that good practice? @AnandBaburajan. Also, the senderID itself is the receiverID in this case, so do we need that redundancy? I guess it is okay to change the interface.

**Comment by PushkarDureja** [comment]
Umm, I dont think sender email will be same as receiver. Receiver will always be the poll creator where as sender email will always be of logged in user and is used as a parameter in nodemailer @AthulJoshi

File: src/routes/v1/pollResponse.ts [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+const router: Router = express.Router();
+
+router.post('/', async (req: Request, res: Response) => {
+  interface Data {
+    pollID: string;
+    pollTitle: string;
+    senderName: string;
+    senderEmailID: string;
+    receiverIDs: string[];
+  }
```
Comment (AthulJoshi) [inline src/routes/v1/pollResponse.ts:17]: We would be making a POST request from the client side, so usually senderName and senderEmailID is passed to be set at the ```from``` field of nodemailer. But in this case, the sender is RocketMeet, so we wont be needing those 2 fields.

File: src/routes/v1/pollResponse.ts [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+    const transporter: Transporter = createTransport({
+      service: 'gmail',
+      auth: {
+        user: email,
+        pass: password,
+      },
+    });
+      const mailOptions = {
+        from: `RocketMeet <${email}>`,
```
Comment (AthulJoshi) [inline src/routes/v1/pollResponse.ts:39]: See this line, and those 2 fields aren't used anywhere, so those 2 wont be needed I guess.

Comment (PushkarDureja) [reply src/routes/v1/pollResponse.ts:39]: If you see line  24, sender's name is used there and about sender email, I thought using it in the `reply to` parameter was the right thing but I was not sure. What do you think?

**Comment by AnandBaburajan** [comment]
I'll take a final look at the code once the doubts here are cleared, but a comment:

> the senderID itself is the receiverID in this case

Since this PR fixes https://github.com/RocketMeet/RocketMeet-client/issues/95 the `senderEmailID` would be voter's email ID and `receiverID` would be the poll creator's email ID. Please let me know if I got something wrong!

File: src/routes/v1/pollResponse.ts (hunk 2 above)
Comment (AthulJoshi) [reply src/routes/v1/pollResponse.ts:39]: Yea, that would be alright

**Comment by AthulJoshi** [comment]
> I'll take a final look at the code once the doubts here are cleared, but a comment:
> 
> > the senderID itself is the receiverID in this case
> 
> Since this PR fixes [RocketMeet/RocketMeet-client#95](https://github.com/RocketMeet/RocketMeet-client/issues/95) the `senderEmailID` would be voter's email ID and `receiverID` would be the poll creator's email ID. Please let me know if I got something wrong!
> 
> Edit:
> 
> > Receiver will always be the poll creator where as sender email will always be of logged in user
> 
> Yep (^_^")

I guess it's ok then, maybe change the receiverIDs array to just receiverID?

**Comment by PushkarDureja** [comment]
@AthulJoshi But for that, We need to create a new Interface which will be used just for this purpose which I dont think is a good approach. Can you suggest any other alternative ?

**Comment by AthulJoshi** [comment]
> @AthulJoshi But for that, We need to create a new Interface which will be used just for this purpose which I dont think is a good approach. Can you suggest any other alternative ?

I don't get it. You have already created an interface here, it's not imported right? So why can't a field be changed? Oh is it because it would cause confusions in the frontend api calls?

**Comment by PushkarDureja** [comment]
Yes, I have used a predefined Interface in frontend, So in order to change a field, I think I have to create a new Interface in frontend 

> > @AthulJoshi But for that, We need to create a new Interface which will be used just for this purpose which I dont think is a good approach. Can you suggest any other alternative ?
> 
> I don't get it. You have already created an interface here, it's not imported right? So why can't a field be changed? Oh is it because it would cause confusions in the frontend api calls?

**Comment by AthulJoshi** [comment]
I guess everything is good then. @AnandBaburajan please have a look .

**Comment by PushkarDureja** [comment]
@AnandBaburajan can you please review it?

File: src/routes/v1/pollResponse.ts [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    pollID: string;
+    pollTitle: string;
+    senderName: string;
+    senderEmailID: string;
+    receiverIDs: string[];
+  }
+
+  const data: Data = req.body;
+
+    const filePath = path.join(__dirname, '../../../template/pollResponse.html');
```
Comment (AnandBaburajan) [inline src/routes/v1/pollResponse.ts:21]: I'm curious on why the auth `data.senderEmailID !== req.currentUser.email` check was removed from here.

**Review by AnandBaburajan — CHANGES_REQUESTED** [review]
I've added one comment, PTAL. Also if you can, try to align your code with the rest of the code. Using Prettier would help I think.

File: src/routes/v1/pollResponse.ts (hunk 3 above)
Comment (PushkarDureja) [reply src/routes/v1/pollResponse.ts:21]: My bad, I will add it back in the next commit along with code alignment.

Comment (PushkarDureja) [reply src/routes/v1/pollResponse.ts:21]: > I've added one comment, PTAL. Also if you can, try to align your code with the rest of the code. Using Prettier would help I think.

I checked my code but didn't find any misalignment, nor did I get any warnings from ES-Lint or prettier. Can you please specify the part where you think the code is unaligned. Sorry if I missed something :(

Comment (AnandBaburajan) [reply src/routes/v1/pollResponse.ts:21]: Check lines 19 - 21. But I think that would be fixed with the auth check. You won't get any warnings for misalignments, but Prettier automatically aligns the code for you (not sure if it needs to be configured to do that).

**Review by AnandBaburajan — APPROVED** [review]
Looks good, thanks, merging :D

**Comment by AnandBaburajan** [comment]
@AthulJoshi please update the API docs too whenever you're free.

**Comment by PushkarDureja** [comment]
> Looks good, thanks, merging :D

Thanks, Great working with you, will surely work on other issues too :)

**Pull request merged**